http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -344,6 +346,7 @@ protected void channelRead0(ChannelHandlerContext ctx, io.micronaut.http.HttpReq
                     .collect(Collectors.toList());
             if (!anyUriRoutes.isEmpty()) {
                 setRouteAttributes(request, anyUriRoutes.get(0));
+                request.setAttribute(AVAILABLE_HTTP_METHODS, anyUriRoutes.stream().map(UriRouteMatch::getHttpMethod).collect(Collectors.toList()));
             }
```

Comment:
This does not consider routes from filters, meaning if you have a filter which handles a route that nothing else does, CORS no longer works with it

Reply:
reported as https://github.com/micronaut-projects/micronaut-core/issues/7934